src/ape/types/__init__.py

```diff
@@ -227,8 +186,47 @@ def get(self, item: str, default: Optional[Any] = None) -> Any:
         return self.event_arguments.get(item, default)
 
 
-class ContractLogEventArguments(BaseModel):
-    pass
+class ContractLog(EventArguments):
```

Comment:
What happens in the `dict` / `JSON` version of this class when there are naming collisions between event arguments and the normal properties on this class (e.g. `contract_address`)?

For example, if I defined a topic named `contract_address` in my contract Event declaration, there would **not** be a way to define both values in a raw dict of a `ContractLog`, thus causing serializations issues with this class.

Or am I misunderstanding?

Reply:
you are absolutely correct here, this is why I'm so stuck in the weeds with this task. I'm doing a really deep recon here, trying to find all possible values, and then I'm trying to figure out how to handle collisions. I'm starting to lean on having 2 `BaseModel`'s and then pulling in `EventArguments` as a pydantic model instead of as a dict like we were before. Then I would still be able to manage the cache database for `ContractEvents`. Idk, I almost want to use a document database for this. It almost doesn't make sense to attempt to flatten this because `event_arguments` are so variable. @fubuloubu what do you think?

Reply:
This is a stale conversation, and we have addressed this topic a few times. I am going to move this to resolved and continue working on this issue here.